sts/client_test.go

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sts
+
+import (
+	"context"
+	"crypto/tls"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/vim25/methods"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+func TestIssueHOK(t *testing.T) {
+	url := os.Getenv("GOVC_TEST_URL")
+	if url == "" {
+		t.SkipNow()
+	}
+
+	u, err := soap.ParseURL(url)
+	if err != nil {
+		t.Fatal(err)
+	}
+	u.User = nil // avoid govmomi.NewClient Login() call
+
+	// solution user must exist, for example run on vCenter VM:
```

Comment:
I found the API to create a solution user, but the sso-adminserver requires SAML token auth.. so creating a solution user depends on this PR.  Will follow up with a separate PR for that to avoid the manual setup for these tests.